File: unit_04/w10_d1/lab/calculatorChall/js/spec/Calculator-spec.js
const Calculator = require('../Calculator');

describe('Calculator', function() {
  let calculator;

  beforeEach(() => {
    calculator = new Calculator();
  });

  describe('#add', () => {
    it('adds two numbers', () => {
      const actual = calculator.add(2, 3);
      const expectation = 5;
      expect(actual).toEqual(expectation);
    });
  });

  describe('with previous answer of 5 found', () => {
    beforeEach(() => {
      calculator.add(2, 3);
    });

    describe('#add', () => {
      it('adds a number to previous answer', () => {
        expect(calculator.add(2)).toEqual(5 + 2);
        expect(calculator.add(2)).toEqual(5 + 2 + 2);
      });
    });

  });
});

File: unit_04/w10_d1/lab/calculatorChall/ruby/spec/calculator_spec.rb
require "./calculator"

RSpec.describe Calculator do
  before(:each) do
    @calculator = Calculator.new
  end

  describe "#add" do
    it "adds two numbers" do
      actual = @calculator.add(2, 3)
      expectation = 5
      expect(actual).to eq(expectation)
    end
  end

  context "with previous answer of 5 found" do
    before(:each) do
      @calculator.add(2, 3)
    end

    describe "#add" do
      it "adds a number to previous answer" do
        expect(@calculator.add(2)).to eq(5 + 2)
        expect(@calculator.add(2)).to eq(5 + 2 + 2)
      end
    end
  end
end
